### DISCUSSION TITLE
Run an INFORM job concurrently with point queuing

### DISCUSSION THREAD

**Discussion by IABSTUDIO** [Yaskawa-Global/motoros2#304] [Q&A] [answered]
Hello,

I'm working on a robot automation project with GP-4 + YRC1000 + Motoros2 humble.
I am trying to simplify making trajectories by replacing some common actions in the entire process with INFORM jobs.
Is there a way to run INFORM jobs stored in robot controller at the same time as using point queue mode with Motoros2?

Thank you.

**Comment by gavanderhoorn** [comment, marked as answer]
> Is there a way to run INFORM jobs stored in robot controller at the same time as using point queue mode with Motoros2?

This is currently not supported.

You can start and stop *queuing mode* using the appropriate services, then interleave running another INFORM job, but you can't run `INIT_ROS` and another INFORM job at the same time -- at least not with the current implementation.

INFORM job CRUD + coordination (ie: job management + start & stop) are on the roadmap, and there is some work towards that (see #278 fi), but this isn't finished yet.

It would also probably not support ROS-controlled motion to happen concurrently with another INFORM job running though. MotoROS2 already *uses* an INFORM job to control motion itself (that's `INIT_ROS`).

**Reply by IABSTUDIO** [reply]
Thanks for your comment. @gavanderhoorn
>INFORM job CRUD + coordination (ie: job management + start & stop) are on the roadmap, and there is some work towards that (see https://github.com/Yaskawa-Global/motoros2/issues/278 fi), but this isn't finished yet.

Hope this to be updated soon!

**Reply by gavanderhoorn** [reply]
Just to emphasise: this would still not allow you to run any other INFORM job concurrently with `INIT_ROS`.

**Reply by IABSTUDIO** [reply]
Okay, I understood.
Then if #278 is finished, can I run specific INFORM job stored in controller after queueing mode is disabled?
Or is there any way to run INFORM job except `INIT_ROS` with current Motoros2 distribution?

**Reply by gavanderhoorn** [reply]
> Then if #278 is finished, can I run specific INFORM job stored in controller after queueing mode is disabled? 

#278 only adds job file management (ie: created, retrieve, update and delete jobs).

It doesn't yet support starting & stopping jobs.

There is some work in that direction ([here](https://github.com/gavanderhoorn/motoros2_interfaces/commit/6df0f769dda75339e9bef579cfa229a4b037cba4) fi), but it's not implemented yet.

> Or is there any way to run INFORM job except `INIT_ROS` with current Motoros2 distribution?

Not directly.

I would have to check, but perhaps @ted-miller knows whether jobs can be started from a MotoPlus application by manipulating IOs on the controller? In that case, you could theoretically start a job by using the `write_*_io` services (ie: `write_single_io`, etc).

**Reply by ted-miller** [reply]
You can't necessarily invoke an arbitrary job from I/O. But if a job is running, you can control the flow and call subjobs based on I/O. 

There's a proof-of-concept implementation available in this discussion: https://github.com/ros-industrial/motoman/discussions/463

I don't really recommend doing this... but it's possible.
